### PR TITLE
docs: update guides proto paths

### LINKED ISSUE #4197: docs: update proto paths
See https://github.com/tendermint/tendermint/issues/4191

We need to update the docs (including tutorials) to reflect the changes made in https://github.com/tendermint/tendermint/pull/4007

### PR DESCRIPTION
**Pull request by marbar3778**
- update proto path of libs/kv for fetching and remove GOPATH from go guides. 
- removed items from makefile that weren't used anymore

closes #4197 

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: docs/guides/go-built-in.md [hunk 1]
```diff
@@ -55,8 +55,8 @@ $ echo $GOPATH
 We'll start by creating a new Go project.
 
 ```sh
-$ mkdir -p $GOPATH/src/github.com/me/kvstore
-$ cd $GOPATH/src/github.com/me/kvstore
+$ mkdir -p kvstore
```
Comment (melekes) [inline docs/guides/go-built-in.md]:
```suggestion
$ mkdir kvstore
```

**Comment by tessr** [comment]
Are the makefile changes also addressing the same issue, or are they taking care of something else?

File: docs/guides/go.md [hunk 2]
```diff
@@ -58,8 +58,8 @@ $ echo $GOPATH
 We'll start by creating a new Go project.
 
 ```sh
-$ mkdir -p $GOPATH/src/github.com/me/kvstore
-$ cd $GOPATH/src/github.com/me/kvstore
+$ mkdir -p kvstore
```
Comment (melekes) [inline docs/guides/go.md]:
```suggestion
$ mkdir kvstore
```

**Comment by marbar3778** [comment]
>re the makefile changes also addressing the same issue, or are they taking care of something else?

its removal of things that aren't present anymore in the repo. It should of been done a while ago, but I missed it when I was transitioning the db things to the tm-db repo.

**Pull request merged**